### ISSUE TITLE
Problem installing ocatave 4.3-d0221e3675ef

### ISSUE THREAD

**Issue by AtheistApe** [dpo/homebrew-openblas#67]
I've followed the instructions at:
http://www.schoeps.org/home/2018/01/how-to-compile-gnu-octave-with-openblas-on-macos/

to install octave but have run into problems. I think the part below is the relevant section; It seems to have something to do with "QButtonGroup".

----------------------------------------------------------------------------------------------------

libgui/src/settings-dialog.cc:299:39: error: allocation of incomplete type 'QButtonGroup'
  QButtonGroup *icon_size_group = new QButtonGroup (this);
                                      ^~~~~~~~~~~~
/usr/local/Cellar/qt/5.11.0/include/QtWidgets/qabstractbutton.h:53:7: note: forward declaration of 'QButtonGroup'
class QButtonGroup;
      ^
libgui/src/settings-dialog.cc:300:18: error: member access into incomplete type 'QButtonGroup'
  icon_size_group->addButton (ui->icon_size_small);
                 ^
/usr/local/Cellar/qt/5.11.0/include/QtWidgets/qabstractbutton.h:53:7: note: forward declaration of 'QButtonGroup'
class QButtonGroup;
      ^
libgui/src/settings-dialog.cc:301:18: error: member access into incomplete type 'QButtonGroup'
  icon_size_group->addButton (ui->icon_size_normal);
                 ^
/usr/local/Cellar/qt/5.11.0/include/QtWidgets/qabstractbutton.h:53:7: note: forward declaration of 'QButtonGroup'
class QButtonGroup;
      ^
libgui/src/settings-dialog.cc:302:18: error: member access into incomplete type 'QButtonGroup'
  icon_size_group->addButton (ui->icon_size_large);
                 ^
/usr/local/Cellar/qt/5.11.0/include/QtWidgets/qabstractbutton.h:53:7: note: forward declaration of 'QButtonGroup'
class QButtonGroup;
      ^
libgui/src/settings-dialog.cc:309:34: error: allocation of incomplete type 'QButtonGroup'
  QButtonGroup *icon_group = new QButtonGroup (this);
                                 ^~~~~~~~~~~~
/usr/local/Cellar/qt/5.11.0/include/QtWidgets/qabstractbutton.h:53:7: note: forward declaration of 'QButtonGroup'
class QButtonGroup;
      ^
libgui/src/settings-dialog.cc:310:13: error: member access into incomplete type 'QButtonGroup'
  icon_group->addButton (ui->general_icon_octave);
            ^
/usr/local/Cellar/qt/5.11.0/include/QtWidgets/qabstractbutton.h:53:7: note: forward declaration of 'QButtonGroup'
class QButtonGroup;
      ^
libgui/src/settings-dialog.cc:311:13: error: member access into incomplete type 'QButtonGroup'
  icon_group->addButton (ui->general_icon_graphic);
            ^
/usr/local/Cellar/qt/5.11.0/include/QtWidgets/qabstractbutton.h:53:7: note: forward declaration of 'QButtonGroup'
class QButtonGroup;
      ^
libgui/src/settings-dialog.cc:312:13: error: member access into incomplete type 'QButtonGroup'
  icon_group->addButton (ui->general_icon_letter);
            ^
/usr/local/Cellar/qt/5.11.0/include/QtWidgets/qabstractbutton.h:53:7: note: forward declaration of 'QButtonGroup'
class QButtonGroup;
      ^
8 errors generated.
make[2]: *** [libgui/src/libgui_src_la-settings-dialog.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2

**Comment by AtheistApe** [comment]
Update:

I was able to complete the install by excluding the '--with-qt' parameter; The command line that compiled was:

brew install dpo/openblas/octave --devel --with-java

**Comment by schoeps** [comment]
Yes, `brew install dpo/openblas/octave --devel --with-java` should work. It does not compile the graphical user interface and thus work also without qt (and its buttons). Do you want the gui or not?

**Comment by AtheistApe** [comment]
Yes, I would like the gui. I've tried brew reinstalling qt and pyqt and tried setting the various paths and environment variables as suggested below without success:

 echo 'export PATH="/usr/local/opt/qt/bin:$PATH"' >> ~/.zshrc

For compilers to find this software you may need to set:
    LDFLAGS:  -L/usr/local/opt/qt/lib
    CPPFLAGS: -I/usr/local/opt/qt/include
For pkg-config to find this software you may need to set:
    PKG_CONFIG_PATH: /usr/local/opt/qt/lib/pkgconfig

**Comment by schoeps** [comment]
I can confirm this issue. Octave seems to have a problem with the new qt 5.11. I will file a bug report at savannah. Sorry, that might need some time until it gets fixed. I cannot help here, I am no qt-guru whatsoever.

**Comment by schoeps** [comment]
Ok, let's try this: first get the most recent formula (this overwrites the previous one, nothing which can't be undone):
`
curl https://raw.githubusercontent.com/schoeps/homebrew-openblas/master/octave.rb -o /usr/local/Homebrew/Library/Taps/dpo/homebrew-openblas/octave.rb 
`

Then try to compile
`
brew install octave --with-qt
`

I am currently compiling; seems to work.

**Comment by AtheistApe** [comment]
Many thanks for your help. While compiling I got a warning message “Some tests failed” but it appears to run fine when I execute it with the —force-gui option. 

Thanks again,
Gary

**Comment by legout** [comment]
Unfortunately i am not able install octave using the most recent formula. 

```
==> Patching
==> Applying download.php
can't find file to patch at input line 4
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff -r bb43ec3a304e libgui/graphics/Figure.cc
|--- a/libgui/graphics/Figure.cc	Thu Apr 26 06:49:06 2018 -0400
|+++ b/libgui/graphics/Figure.cc	Thu Apr 26 17:46:49 2018 -0700
--------------------------
No file to patch.  Skipping patch.
3 out of 3 hunks ignored
can't find file to patch at input line 68
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff -r bb43ec3a304e libgui/graphics/Figure.h
|--- a/libgui/graphics/Figure.h	Thu Apr 26 06:49:06 2018 -0400
|+++ b/libgui/graphics/Figure.h	Thu Apr 26 17:46:49 2018 -0700
--------------------------
No file to patch.  Skipping patch.
2 out of 2 hunks ignored
can't find file to patch at input line 88
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff -r bb43ec3a304e libgui/graphics/FigureWindow.cc
|--- a/libgui/graphics/FigureWindow.cc	Thu Apr 26 06:49:06 2018 -0400
|+++ b/libgui/graphics/FigureWindow.cc	Thu Apr 26 17:46:49 2018 -0700
--------------------------
No file to patch.  Skipping patch.
1 out of 1 hunk ignored
can't find file to patch at input line 102
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff -r bb43ec3a304e libgui/graphics/FigureWindow.h
|--- a/libgui/graphics/FigureWindow.h	Thu Apr 26 06:49:06 2018 -0400
|+++ b/libgui/graphics/FigureWindow.h	Thu Apr 26 17:46:49 2018 -0700
--------------------------
No file to patch.  Skipping patch.
1 out of 1 hunk ignored
can't find file to patch at input line 124
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff -r bb43ec3a304e libgui/graphics/GLCanvas.cc
|--- a/libgui/graphics/GLCanvas.cc	Thu Apr 26 06:49:06 2018 -0400
|+++ b/libgui/graphics/GLCanvas.cc	Thu Apr 26 17:46:49 2018 -0700
--------------------------
No file to patch.  Skipping patch.
2 out of 2 hunks ignored
can't find file to patch at input line 151
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff -r bb43ec3a304e libinterp/corefcn/gl-render.cc
|--- a/libinterp/corefcn/gl-render.cc	Thu Apr 26 06:49:06 2018 -0400
|+++ b/libinterp/corefcn/gl-render.cc	Thu Apr 26 17:46:49 2018 -0700
--------------------------
No file to patch.  Skipping patch.
7 out of 7 hunks ignored
can't find file to patch at input line 233
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff -r bb43ec3a304e libinterp/corefcn/gl-render.h
|--- a/libinterp/corefcn/gl-render.h	Thu Apr 26 06:49:06 2018 -0400
|+++ b/libinterp/corefcn/gl-render.h	Thu Apr 26 17:46:49 2018 -0700
--------------------------
No file to patch.  Skipping patch.
1 out of 1 hunk ignored
can't find file to patch at input line 246
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff -r bb43ec3a304e libinterp/corefcn/graphics.cc
|--- a/libinterp/corefcn/graphics.cc	Thu Apr 26 06:49:06 2018 -0400
|+++ b/libinterp/corefcn/graphics.cc	Thu Apr 26 17:46:49 2018 -0700
--------------------------
No file to patch.  Skipping patch.
8 out of 8 hunks ignored
can't find file to patch at input line 354
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff -r bb43ec3a304e libinterp/corefcn/graphics.in.h
|--- a/libinterp/corefcn/graphics.in.h	Thu Apr 26 06:49:06 2018 -0400
|+++ b/libinterp/corefcn/graphics.in.h	Thu Apr 26 17:46:49 2018 -0700
--------------------------
No file to patch.  Skipping patch.
1 out of 1 hunk ignored
can't find file to patch at input line 365
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff -r bb43ec3a304e m4/acinclude.m4
|--- a/m4/acinclude.m4	Thu Apr 26 06:49:06 2018 -0400
|+++ b/m4/acinclude.m4	Thu Apr 26 17:46:49 2018 -0700
--------------------------
No file to patch.  Skipping patch.
2 out of 2 hunks ignored
can't find file to patch at input line 413
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff -r bb43ec3a304e scripts/plot/util/print.m
|--- a/scripts/plot/util/print.m	Thu Apr 26 06:49:06 2018 -0400
|+++ b/scripts/plot/util/print.m	Thu Apr 26 17:46:49 2018 -0700
--------------------------
No file to patch.  Skipping patch.
1 out of 1 hunk ignored
Error: Failure while executing: patch -g 0 -f -p1 -i /private/tmp/octave--patch-20180714-19514-mhkz3h/download.php
```

**Comment by orqldon** [comment]
I have the exact same error as legout.  I'm running High Sierra 10.13.6 with the latest Xcode and command line tools.

I used 
`curl https://raw.githubusercontent.com/schoeps/homebrew-openblas/master/octave.rb -o /usr/local/Homebrew/Library/Taps/dpo/homebrew-openblas/octave.rb`
followed by
`brew install octave --with-qt --with-java`

any ideas @schoeps? or @legout did you find a solution?  Thanks!

**Comment by schoeps** [comment]
@orqldon @legout I just tried to reproduce the failure but I can't. It just works for me. Can you upload your log files? (see https://docs.brew.sh/Troubleshooting and in particular: `brew gist-logs octave`)

**Comment by legout** [comment]
@schoeps The logs are here:
https://gist.github.com/legout/3bce108d4f3539e2774ea4ea1aeefa40

**Comment by schoeps** [comment]
@legout Strangely there is no output from configure or make. If this does not work, can you try this: 
```
# get newest formula
brew update
brew upgrade
# make sure that dependencies are compiled from dpo's repository
brew reinstall openblas
brew reinstall dpo/homebrew-openblas/arpack
brew reinstall dpo/homebrew-openblas/qrupdate
brew reinstall dpo/homebrew-openblas/suite-sparse
brew reinstall dpo/homebrew-openblas/sundials27
#compile octave
brew install dpo/homebrew-openblas/octave --verbose --with-qt --with-java
```
and copy the output here?

**Comment by legout** [comment]
@schoeps 
Now i was able to install octave using the guidance above. Thanks!

**Comment by bobbydas** [comment]
No joy :( Still error building with your updated instructions. 
https://gist.github.com/bobbydas/7e58cb04f21f8903f1e123fe72922f7e

**Comment by jbenzakoun** [comment]
@schoeps i had the exact same problem when installing octave with qt 5.11
I managed to find a solution (see https://bugs.gentoo.org/655844). Just add the following include line to **libgui/src/settings-dialog.cc**
`#include <QButtonGroup>`

**Comment by schoeps** [comment]
@jbenzakoun yes, thanks! It's bug #53978 and it's also fixed in the pull request https://github.com/dpo/homebrew-openblas/pull/61/files
